h/static/scripts/group-service.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * @ngdoc service
+ * @name  group
+ *
+ * @description
+ * Get and set the UI's currently focused group.
+ */
+'use strict';
+
+// @ngInject
+function group(session) {
+  // The currently focused group. This is the group that's shown as selected in
+  // the groups dropdown, the annotations displayed are filtered to only ones
+  // that belong to this group, and any new annotations that the user creates
+  // will be created in this group.
+  var focusedGroup;
+
+  // The public "group" is always available, regardless of what real groups (if
+  // any) the user is a member of.
+  var publicGroup = {
+    name: 'Public',
+    // We need a sentinel value to identify the public group (which is not
+    // really a group). '__public__' won't clash with any of the hashids that
+    // we use to identify real groups because the hashids never contain _.
+    hashid: '__public__'
+  };
+
+  // Return the list of available groups.
+  var groups = function() {
+    return [publicGroup].concat(session.state.groups || []);
```

Comment:
Can we just get the public group into the session group list and then this doesn't need to be a function but just a property? Even if it meant an extra web request maybe we should mutate the current group in the session and just use that as the source of all this information. It would mean the privacy directive is a bit clearer, too, because it wouldn't need to use a getter to fetch the current group, it could just be a binding. Maybe.